compiler/test/suites/chars.re

```diff
@@ -56,4 +83,28 @@ describe("chars", ({test}) => {
     "let x = '\\u{110000}'",
     "Illegal unicode code point",
   );
+  // parse locations
+  assertParseWithLocs(
+    "char_loc_simple",
+    "'a'",
+    {
+      statements: [
+        char(~loc=mk_loc("char_loc_simple", (1, 0, 0), (1, 3, 0)), "a"),
+      ],
+      comments: [],
+      prog_loc: mk_loc("char_loc_simple", (1, 0, 0), (1, 3, 0)),
+    },
+  );
+  // note that the char length is calculated as having 4 bytes, not 1 codepoint
```

Comment:
This actually reminds me of that question I had around this—how do tools like vscode handle byte-based locs rather than unicode chars? 🤔

Reply:
🤷‍♂️ no clue. Want to make a follow-up issue?

Reply:
No 😛 

Sorta like a "if/when we see it's a problem..." kinda feeling.

Reply:
But! That does make me feel like we should have a test for a char like `'\u{1F3F4}'` too.

Reply:
Added.

Reply:
Nice.